src/Rules/Honeypot.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace AntispamBee\Rules;
+
+use AntispamBee\Helpers\Honeypot as HoneypotField;
+use AntispamBee\Helpers\ContentTypeHelper;
+use AntispamBee\Helpers\DataHelper;
+use AntispamBee\Helpers\DebugMode;
+use AntispamBee\Helpers\Settings;
+use AntispamBee\Interfaces\SpamReason;
+
+/**
+ * Adds honeypot to comment form and checks if it is filled.
+ */
+class Honeypot extends ControllableBase implements SpamReason {
+	protected static $slug = 'asb-honeypot';
+	protected static $supported_types = [ ContentTypeHelper::COMMENT_TYPE ];
+
+	/**
+	 * Initialize the rule.
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		add_filter( 'antispam_bee_rules', [ __CLASS__, 'add_rule' ] );
+
+		add_filter(
+			'comment_form_field_comment',
+			function ( $field_markup ) {
+				if ( ! static::is_active( ContentTypeHelper::COMMENT_TYPE ) ) {
+					return $field_markup;
+				}
+
+				return HoneypotField::inject( $field_markup, [ 'field_id' => 'comment' ] );
+			}
+		);
+	}
+
+	public static function verify( $item ) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing
+		if ( isset( $_POST['ab_spam__hidden_field'] ) && 1 === $_POST['ab_spam__hidden_field'] ) {
+			return 999;
+		}
+
+		return 0;
+	}
+
+	public static function precheck() {
+		// phpcs:disable WordPress.Security.NonceVerification.Missing
+		if ( is_feed() || is_trackback() || empty( $_POST ) ) {
+			return;
+		}
+		
+		$request_uri  = Settings::get_key( $_SERVER, 'REQUEST_URI' );
+		$request_path = DataHelper::parse_url( $request_uri, 'path' );
+
+		if ( strpos( $request_path, 'wp-comments-post.php' ) === false ) {
+			return;
+		}
+		$fields = [];
+		foreach ( $_POST as $key => $value ) {
+			if ( isset( $fields['plugin_field'] ) ) {
```

Comment:
Here we rely on the order of the honeypot and the field that is replaced by it. We should not do that, and find a way to handle that independently from the order in `$_POST`